unexpected EOF while creating heroku app with empty config vars

We're trying to create a heroku app using the heroku provider. When the config vars are an empty set the heroku app is created, but it's not stored in the tfstate file and terraform crashes. 

Everything works again if the empty config vars block is removed.

crash log below. 


2015/03/16 12:44:49 [INFO] Terraform version: 0.3.7  0bc0c03fece07c4f21ee5195743116a4d418f234+CHANGES
2015/03/16 12:44:49 Detected home directory from env var: /Users/woodhull
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: atlas = /usr/local/terraform/terraform-provider-atlas
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: aws = /usr/local/terraform/terraform-provider-aws
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: cloudflare = /usr/local/terraform/terraform-provider-cloudflare
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: cloudstack = /usr/local/terraform/terraform-provider-cloudstack
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: consul = /usr/local/terraform/terraform-provider-consul
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: digitalocean = /usr/local/terraform/terraform-provider-digitalocean
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: dnsimple = /usr/local/terraform/terraform-provider-dnsimple
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: google = /usr/local/terraform/terraform-provider-google
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: heroku = /usr/local/terraform/terraform-provider-heroku
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: mailgun = /usr/local/terraform/terraform-provider-mailgun
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: null = /usr/local/terraform/terraform-provider-null
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: file = /usr/local/terraform/terraform-provisioner-file
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: local-exec = /usr/local/terraform/terraform-provisioner-local-exec
2015/03/16 12:44:49 [DEBUG] Discoverd plugin: remote-exec = /usr/local/terraform/terraform-provisioner-remote-exec
2015/03/16 12:44:49 Detected home directory from env var: /Users/woodhull
2015/03/16 12:44:49 [DEBUG] Attempting to open CLI config file: /Users/woodhull/.terraformrc
2015/03/16 12:44:49 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2015/03/16 12:44:49 [DEBUG] Creating graph for path: [root]
2015/03/16 12:44:49 [DEBUG] Starting plugin: /usr/local/terraform/terraform-provider-heroku []string{"/usr/local/terraform/terraform-provider-heroku"}
2015/03/16 12:44:49 [DEBUG] Waiting for RPC address for: /usr/local/terraform/terraform-provider-heroku
2015/03/16 12:44:49 Detected home directory from env var: /Users/woodhull
2015/03/16 12:44:49 terraform-provider-heroku: 2015/03/16 12:44:49 Plugin address: unix /var/folders/7l/0mdnrbyn0jz9yp361n6j7y7h0000gn/T/tf-plugin371916803
2015/03/16 12:44:50 [DEBUG] Starting plugin: /usr/local/terraform/terraform-provider-aws []string{"/usr/local/terraform/terraform-provider-aws"}
2015/03/16 12:44:50 [DEBUG] Waiting for RPC address for: /usr/local/terraform/terraform-provider-aws
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 Plugin address: unix /var/folders/7l/0mdnrbyn0jz9yp361n6j7y7h0000gn/T/tf-plugin800872214
2015/03/16 12:44:50 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/03/16 12:44:50 [DEBUG] Creating graph for path: [root]
2015/03/16 12:44:50 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/03/16 12:44:50 [INFO] Validating provider: aws
2015/03/16 12:44:50 [INFO] Validating provider: heroku
2015/03/16 12:44:50 [DEBUG] heroku_app.default: expanding to count = 1
2015/03/16 12:44:50 [INFO] Validating resource: heroku_app.default
2015/03/16 12:44:50 [DEBUG] aws_security_group.allow_pg_from_anywhere: expanding to count = 1
2015/03/16 12:44:50 [INFO] Validating resource: aws_security_group.allow_pg_from_anywhere
2015/03/16 12:44:50 [DEBUG] aws_db_instance.default: expanding to count = 1
2015/03/16 12:44:50 [INFO] Validating resource: aws_db_instance.default
2015/03/16 12:44:50 [DEBUG] Creating graph for path: [root]
2015/03/16 12:44:50 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/03/16 12:44:50 [INFO] Configuring provider: aws
2015/03/16 12:44:50 [INFO] Configuring provider: heroku
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Building AWS auth structure
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Building AWS region structure
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing EC2 connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing ELB connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing AutoScaling connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing S3 connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing RDS connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing Route53 connection
2015/03/16 12:44:50 [DEBUG] aws_security_group.allow_pg_from_anywhere: expanding to count = 1
2015/03/16 12:44:50 [INFO] Module root walking: aws_security_group.allow_pg_from_anywhere (Graph node: aws_security_group.allow_pg_from_anywhere)
2015/03/16 12:44:50 terraform-provider-heroku: 2015/03/16 12:44:50 [INFO] Initializing Heroku client
2015/03/16 12:44:50 terraform-provider-heroku: 2015/03/16 12:44:50 [INFO] Heroku Client configured for user: heroku@controlshiftlabs.com
2015/03/16 12:44:50 [DEBUG] heroku_app.default: expanding to count = 1
2015/03/16 12:44:50 [INFO] Module root walking: heroku_app.default (Graph node: heroku_app.default)
2015/03/16 12:44:50 [DEBUG] heroku_app.default: Not refreshing, ID is empty
2015/03/16 12:44:50 [DEBUG] aws_db_instance.default: expanding to count = 1
2015/03/16 12:44:50 [INFO] Module root walking: aws_db_instance.default (Graph node: aws_db_instance.default)
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [DEBUG] DB Instance describe configuration: rds.DescribeDBInstances{DBInstanceIdentifier:"agra-staging-rds"}
2015/03/16 12:44:50 [DEBUG] Creating graph for path: [root]
2015/03/16 12:44:50 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/03/16 12:44:50 [INFO] Configuring provider: aws
2015/03/16 12:44:50 [INFO] Configuring provider: heroku
2015/03/16 12:44:50 terraform-provider-heroku: 2015/03/16 12:44:50 [INFO] Initializing Heroku client
2015/03/16 12:44:50 terraform-provider-heroku: 2015/03/16 12:44:50 [INFO] Heroku Client configured for user: heroku@controlshiftlabs.com
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Building AWS auth structure
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Building AWS region structure
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing EC2 connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing ELB connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing AutoScaling connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing S3 connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing RDS connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing Route53 connection
2015/03/16 12:44:50 [DEBUG] aws_security_group.allow_pg_from_anywhere: expanding to count = 1
2015/03/16 12:44:50 [DEBUG] heroku_app.default: expanding to count = 1
2015/03/16 12:44:50 [INFO] Module root walking: aws_security_group.allow_pg_from_anywhere (Graph node: aws_security_group.allow_pg_from_anywhere)
2015/03/16 12:44:50 [DEBUG] aws_security_group.allow_pg_from_anywhere: Executing diff
2015/03/16 12:44:50 [INFO] Module root walking: heroku_app.default (Graph node: heroku_app.default)
2015/03/16 12:44:50 [DEBUG] heroku_app.default: Executing diff
2015/03/16 12:44:50 [DEBUG] aws_db_instance.default: expanding to count = 1
2015/03/16 12:44:50 [INFO] Module root walking: aws_db_instance.default (Graph node: aws_db_instance.default)
2015/03/16 12:44:50 [DEBUG] aws_db_instance.default: Executing diff
2015/03/16 12:44:50 [DEBUG] heroku_app.default: Diff: &terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"region":*terraform.ResourceAttrDiff{Old:"", New:"us", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x0}, "all_config_vars.#":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "web_url":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "config_vars.#":*terraform.ResourceAttrDiff{Old:"", New:"1", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "id":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x2}, "heroku_hostname":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "git_url":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "name":*terraform.ResourceAttrDiff{Old:"", New:"changesprout-agra-staging", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "stack":*terraform.ResourceAttrDiff{Old:"", New:"cedar-14", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x0}}, Destroy:false, DestroyTainted:false}
2015/03/16 12:44:50 [INFO] Apply walk starting
2015/03/16 12:44:50 [DEBUG] Creating graph for path: [root]
2015/03/16 12:44:50 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/03/16 12:44:50 [INFO] Configuring provider: aws
2015/03/16 12:44:50 [INFO] Configuring provider: heroku
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Building AWS auth structure
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Building AWS region structure
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing EC2 connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing ELB connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing AutoScaling connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing S3 connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing RDS connection
2015/03/16 12:44:50 terraform-provider-aws: 2015/03/16 12:44:50 [INFO] Initializing Route53 connection
2015/03/16 12:44:50 terraform-provider-heroku: 2015/03/16 12:44:50 [INFO] Initializing Heroku client
2015/03/16 12:44:50 terraform-provider-heroku: 2015/03/16 12:44:50 [INFO] Heroku Client configured for user: heroku@controlshiftlabs.com
2015/03/16 12:44:50 [DEBUG] aws_security_group.allow_pg_from_anywhere: expanding to count = 1
2015/03/16 12:44:50 [INFO] Module root walking: aws_security_group.allow_pg_from_anywhere (Graph node: aws_security_group.allow_pg_from_anywhere)
2015/03/16 12:44:50 [DEBUG] aws_security_group.allow_pg_from_anywhere: Diff is empty. Will not apply.
2015/03/16 12:44:50 [DEBUG] aws_db_instance.default: expanding to count = 1
2015/03/16 12:44:50 [INFO] Module root walking: aws_db_instance.default (Graph node: aws_db_instance.default)
2015/03/16 12:44:50 [DEBUG] aws_db_instance.default: Diff is empty. Will not apply.
2015/03/16 12:44:50 [DEBUG] heroku_app.default: expanding to count = 1
2015/03/16 12:44:50 [INFO] Module root walking: heroku_app.default (Graph node: heroku_app.default)
2015/03/16 12:44:50 [DEBUG] heroku_app.default: Executing Apply
2015/03/16 12:44:50 terraform-provider-heroku: 2015/03/16 12:44:50 [DEBUG] App name: changesprout-agra-staging
2015/03/16 12:44:50 terraform-provider-heroku: 2015/03/16 12:44:50 [DEBUG] App region: us
2015/03/16 12:44:50 terraform-provider-heroku: 2015/03/16 12:44:50 [DEBUG] App stack: cedar-14
2015/03/16 12:44:50 terraform-provider-heroku: 2015/03/16 12:44:50 [DEBUG] Creating Heroku app...
2015/03/16 12:44:51 terraform-provider-heroku: 2015/03/16 12:44:51 [INFO] App ID: changesprout-agra-staging
2015/03/16 12:44:51 terraform-provider-heroku: panic: interface conversion: interface is nil, not map[string]interface {}
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 43 [running]:
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/builtin/providers/heroku.updateConfigVars(0xc208253340, 0x19, 0xc208045378, 0x0, 0x0, 0x0, 0xc208379c10, 0x1, 0x1, 0x0, ...)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/builtin/providers/heroku/resource_heroku_app.go:366 +0x293
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/builtin/providers/heroku.resourceHerokuAppCreate(0xc2080f8d80, 0x550320, 0xc208045378, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/builtin/providers/heroku/resource_heroku_app.go:170 +0x92a
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/builtin/providers/heroku.switchHerokuAppCreate(0xc2080f8d80, 0x550320, 0xc208045378, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/builtin/providers/heroku/resource_heroku_app.go:134 +0x14e
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc20810d650, 0xc208121f60, 0xc20812cd10, 0x550320, 0xc208045378, 0x1, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/helper/schema/resource.go:115 +0x2c1
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc20810d9e0, 0xc208128080, 0xc208121f60, 0xc20812cd10, 0x3ca920, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/helper/schema/provider.go:154 +0x1da
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.(*ResourceProviderServer).Apply(0xc2080bef20, 0xc208121a20, 0xc20812ce30, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/resource_provider.go:319 +0x7f
2015/03/16 12:44:51 terraform-provider-heroku: reflect.Value.call(0x47a720, 0x520260, 0x13, 0x5548d0, 0x4, 0xc20815df28, 0x3, 0x3, 0x0, 0x0, ...)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/reflect/value.go:419 +0x10e5
2015/03/16 12:44:51 terraform-provider-heroku: reflect.Value.Call(0x47a720, 0x520260, 0x13, 0xc20815df28, 0x3, 0x3, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/reflect/value.go:296 +0xbc
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*service).call(0xc208117800, 0xc2081177c0, 0xc20810bf10, 0xc2080b2900, 0xc2080bf720, 0x3ca8c0, 0xc208121a20, 0x16, 0x3ca920, 0xc20812ce30, ...)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:382 +0x1f7
2015/03/16 12:44:51 terraform-provider-heroku: created by net/rpc.(*Server).ServeCodec
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:476 +0x44a
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 1 [IO wait]:
2015/03/16 12:44:51 terraform-provider-heroku: net.(*pollDesc).Wait(0xc2080101b0, 0x72, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/fd_poll_runtime.go:84 +0x47
2015/03/16 12:44:51 terraform-provider-heroku: net.(*pollDesc).WaitRead(0xc2080101b0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/fd_poll_runtime.go:89 +0x43
2015/03/16 12:44:51 terraform-provider-heroku: net.(*netFD).accept(0xc208010150, 0x0, 0xa11d48, 0xc20800b008)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/fd_unix.go:419 +0x40b
2015/03/16 12:44:51 terraform-provider-heroku: net.(*UnixListener).AcceptUnix(0xc20801ec40, 0x35670, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/unixsock_posix.go:282 +0x56
2015/03/16 12:44:51 terraform-provider-heroku: net.(*UnixListener).Accept(0xc20801ec40, 0x0, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/unixsock_posix.go:293 +0x4c
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.(*Server).Accept(0xc20800af00, 0xa124c8, 0xc20801ec40)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:33 +0x37
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/plugin.Serve(0xc20800aef0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/plugin/server.go:88 +0x7e8
2015/03/16 12:44:51 terraform-provider-heroku: main.main()
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/builtin/bins/provider-heroku/main.go:11 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 5 [syscall]:
2015/03/16 12:44:51 terraform-provider-heroku: os/signal.loop()
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/os/signal/signal_unix.go:21 +0x1f
2015/03/16 12:44:51 terraform-provider-heroku: created by os/signal.init·1
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/os/signal/signal_unix.go:27 +0x35
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 7 [chan receive]:
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/plugin.func·007()
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/plugin/server.go:79 +0x69
2015/03/16 12:44:51 terraform-provider-heroku: created by github.com/hashicorp/terraform/plugin.Serve
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/plugin/server.go:85 +0x7c4
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 8 [select]:
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/yamux.(*Stream).Read(0xc20805c3c0, 0xc20808a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/stream.go:125 +0x3ab
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).fill(0xc2080585a0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:97 +0x1ce
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).Read(0xc2080585a0, 0xc20800b0a0, 0x1, 0x9, 0x2592, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:174 +0x26c
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadAtLeast(0xa13620, 0xc2080585a0, 0xc20800b0a0, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:298 +0xf1
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadFull(0xa13620, 0xc2080585a0, 0xc20800b0a0, 0x1, 0x9, 0x2c, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:316 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.decodeUintReader(0xa13620, 0xc2080585a0, 0xc20800b0a0, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decode.go:121 +0x99
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).recvMessage(0xc2080b2580, 0xc2080017a0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:76 +0x55
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).decodeTypeSequence(0xc2080b2580, 0xc20801f300, 0x16)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:140 +0x47
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).DecodeValue(0xc2080b2580, 0x3d1d60, 0xc20801f320, 0x16, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:208 +0x192
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).Decode(0xc2080b2580, 0x3d1d60, 0xc20801f320, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:185 +0x297
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc2080de030, 0xc20801f320, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:402 +0x5e
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).readRequestHeader(0xc20803d880, 0xa13780, 0xc2080de030, 0x0, 0x0, 0xc20801f320, 0xc20800b000, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:575 +0x97
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).readRequest(0xc20803d880, 0xa13780, 0xc2080de030, 0xc2080b2480, 0xc20801eda0, 0x3eca00, 0x0, 0x0, 0x0, 0x0, ...)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:542 +0xa2
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).ServeCodec(0xc20803d880, 0xa13780, 0xc2080de030)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:461 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).ServeConn(0xc20803d880, 0xa136a8, 0xc20805c3c0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:453 +0x208
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.(*Server).ServeConn(0xc20800af00, 0xa135c0, 0xc208044078)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:76 +0x530
2015/03/16 12:44:51 terraform-provider-heroku: created by github.com/hashicorp/terraform/rpc.(*Server).Accept
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:39 +0x173
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 9 [IO wait]:
2015/03/16 12:44:51 terraform-provider-heroku: net.(*pollDesc).Wait(0xc208010290, 0x72, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/fd_poll_runtime.go:84 +0x47
2015/03/16 12:44:51 terraform-provider-heroku: net.(*pollDesc).WaitRead(0xc208010290, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/fd_poll_runtime.go:89 +0x43
2015/03/16 12:44:51 terraform-provider-heroku: net.(*netFD).Read(0xc208010230, 0xc2080b4000, 0x1000, 0x1000, 0x0, 0xa11d48, 0xc20812c958)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/fd_unix.go:242 +0x40f
2015/03/16 12:44:51 terraform-provider-heroku: net.(*conn).Read(0xc208044078, 0xc2080b4000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/net.go:121 +0xdc
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).fill(0xc208058240)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:97 +0x1ce
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).Read(0xc208058240, 0xc20800b020, 0xc, 0xc, 0xc20812c9e0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:174 +0x26c
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadAtLeast(0xa13620, 0xc208058240, 0xc20800b020, 0xc, 0xc, 0xc, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:298 +0xf1
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadFull(0xa13620, 0xc208058240, 0xc20800b020, 0xc, 0xc, 0xc, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:316 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/yamux.(*Session).recv(0xc20804cd20)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:340 +0xcb
2015/03/16 12:44:51 terraform-provider-heroku: created by github.com/hashicorp/yamux.newSession
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:93 +0x3f0
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 10 [select]:
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/yamux.(*Session).send(0xc20804cd20)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:298 +0x554
2015/03/16 12:44:51 terraform-provider-heroku: created by github.com/hashicorp/yamux.newSession
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:94 +0x40a
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 11 [select]:
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/yamux.(*Session).keepalive(0xc20804cd20)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:254 +0x106
2015/03/16 12:44:51 terraform-provider-heroku: created by github.com/hashicorp/yamux.newSession
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:96 +0x435
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 13 [select]:
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/yamux.(*Session).AcceptStream(0xc20804cd20, 0xc200000005, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:162 +0x173
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.(*muxBroker).Run(0xc20801ed60)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/mux_broker.go:107 +0x37
2015/03/16 12:44:51 terraform-provider-heroku: created by github.com/hashicorp/terraform/rpc.(*Server).ServeConn
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:65 +0x388
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 15 [select]:
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/yamux.(*Stream).Read(0xc20805c140, 0xc2080b8000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/stream.go:125 +0x3ab
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).fill(0xc208058c60)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:97 +0x1ce
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).Read(0xc208058c60, 0xc20800b4e0, 0x1, 0x9, 0xc20806f858, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:174 +0x26c
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadAtLeast(0xa13620, 0xc208058c60, 0xc20800b4e0, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:298 +0xf1
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadFull(0xa13620, 0xc208058c60, 0xc20800b4e0, 0x1, 0x9, 0x1, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:316 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.decodeUintReader(0xa13620, 0xc208058c60, 0xc20800b4e0, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decode.go:121 +0x99
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).recvMessage(0xc2080b2780, 0xc2080645a0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:76 +0x55
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).decodeTypeSequence(0xc2080b2780, 0xc20801f300, 0x16)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:140 +0x47
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).DecodeValue(0xc2080b2780, 0x3d1d60, 0xc20801f380, 0x16, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:208 +0x192
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).Decode(0xc2080b2780, 0x3d1d60, 0xc20801f380, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:185 +0x297
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc2080def00, 0xc20801f380, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:402 +0x5e
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).readRequestHeader(0xc2080a6080, 0xa13780, 0xc2080def00, 0x0, 0x0, 0xc20801f380, 0xc20800b500, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:575 +0x97
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).readRequest(0xc2080a6080, 0xa13780, 0xc2080def00, 0xc2080b2300, 0xc20801fa00, 0x3caaa0, 0x0, 0x0, 0x0, 0x0, ...)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:542 +0xa2
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).ServeCodec(0xc2080a6080, 0xa13780, 0xc2080def00)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:461 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).ServeConn(0xc2080a6080, 0xa136a8, 0xc20805c140)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:453 +0x208
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.serve(0xa136a8, 0xc20805c140, 0x5ac490, 0x10, 0x5201c0, 0xc20801f2c0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:146 +0x1f9
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.func·001()
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:102 +0x265
2015/03/16 12:44:51 terraform-provider-heroku: created by github.com/hashicorp/terraform/rpc.(*dispenseServer).ResourceProvider
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:103 +0x107
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 21 [select]:
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/yamux.(*Stream).Read(0xc20805c500, 0xc2080f0000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/stream.go:125 +0x3ab
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).fill(0xc2080598c0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:97 +0x1ce
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).Read(0xc2080598c0, 0xc2080ec0c0, 0x1, 0x9, 0xc208071858, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:174 +0x26c
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadAtLeast(0xa13620, 0xc2080598c0, 0xc2080ec0c0, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:298 +0xf1
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadFull(0xa13620, 0xc2080598c0, 0xc2080ec0c0, 0x1, 0x9, 0x1, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:316 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.decodeUintReader(0xa13620, 0xc2080598c0, 0xc2080ec0c0, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decode.go:121 +0x99
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).recvMessage(0xc2080b2d80, 0xc2080646c0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:76 +0x55
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).decodeTypeSequence(0xc2080b2d80, 0xc2080bea00, 0x16)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:140 +0x47
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).DecodeValue(0xc2080b2d80, 0x3d1d60, 0xc2080bea00, 0x16, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:208 +0x192
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).Decode(0xc2080b2d80, 0x3d1d60, 0xc2080bea00, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:185 +0x297
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc2080bccc0, 0xc2080bea00, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:402 +0x5e
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).readRequestHeader(0xc2080a6e40, 0xa13780, 0xc2080bccc0, 0x0, 0x0, 0xc2080bea00, 0xc2080ec000, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:575 +0x97
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).readRequest(0xc2080a6e40, 0xa13780, 0xc2080bccc0, 0xc2080b2d00, 0xc2080be460, 0x3cac80, 0x0, 0x0, 0x0, 0x0, ...)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:542 +0xa2
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).ServeCodec(0xc2080a6e40, 0xa13780, 0xc2080bccc0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:461 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).ServeConn(0xc2080a6e40, 0xa136a8, 0xc20805c500)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:453 +0x208
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.serve(0xa136a8, 0xc20805c500, 0x5ac490, 0x10, 0x5201c0, 0xc2080be440)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:146 +0x1f9
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.func·001()
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:102 +0x265
2015/03/16 12:44:51 terraform-provider-heroku: created by github.com/hashicorp/terraform/rpc.(*dispenseServer).ResourceProvider
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:103 +0x107
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 27 [select]:
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/yamux.(*Stream).Read(0xc20805c640, 0xc2080fa000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/stream.go:125 +0x3ab
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).fill(0xc2080f8000)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:97 +0x1ce
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).Read(0xc2080f8000, 0xc2080ed410, 0x1, 0x9, 0x3f2840, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:174 +0x26c
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadAtLeast(0xa13620, 0xc2080f8000, 0xc2080ed410, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:298 +0xf1
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadFull(0xa13620, 0xc2080f8000, 0xc2080ed410, 0x1, 0x9, 0x1, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:316 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.decodeUintReader(0xa13620, 0xc2080f8000, 0xc2080ed410, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decode.go:121 +0x99
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).recvMessage(0xc2080b3300, 0xc208064ea0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:76 +0x55
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).decodeTypeSequence(0xc2080b3300, 0xc2080bfb00, 0x16)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:140 +0x47
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).DecodeValue(0xc2080b3300, 0x3d1d60, 0xc2080bfb40, 0x16, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:208 +0x192
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).Decode(0xc2080b3300, 0x3d1d60, 0xc2080bfb40, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:185 +0x297
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc2080f4a80, 0xc2080bfb40, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:402 +0x5e
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).readRequestHeader(0xc2080f6280, 0xa13780, 0xc2080f4a80, 0x0, 0x0, 0xc2080bfb40, 0xc2080ed400, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:575 +0x97
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).readRequest(0xc2080f6280, 0xa13780, 0xc2080f4a80, 0xc2080b2f80, 0xc2080fc0e0, 0x513a40, 0x0, 0x0, 0x0, 0x0, ...)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:542 +0xa2
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).ServeCodec(0xc2080f6280, 0xa13780, 0xc2080f4a80)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:461 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).ServeConn(0xc2080f6280, 0xa136a8, 0xc20805c640)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:453 +0x208
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.serve(0xa136a8, 0xc20805c640, 0x5ac490, 0x10, 0x5201c0, 0xc2080bfb20)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:146 +0x1f9
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.func·001()
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:102 +0x265
2015/03/16 12:44:51 terraform-provider-heroku: created by github.com/hashicorp/terraform/rpc.(*dispenseServer).ResourceProvider
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:103 +0x107
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 32 [select]:
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/yamux.(*Stream).Read(0xc20805c780, 0xc208110000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/stream.go:125 +0x3ab
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).fill(0xc2080f8780)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:97 +0x1ce
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).Read(0xc2080f8780, 0xc20810a010, 0x1, 0x9, 0xc208103858, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:174 +0x26c
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadAtLeast(0xa13620, 0xc2080f8780, 0xc20810a010, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:298 +0xf1
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadFull(0xa13620, 0xc2080f8780, 0xc20810a010, 0x1, 0x9, 0x1, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:316 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.decodeUintReader(0xa13620, 0xc2080f8780, 0xc20810a010, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decode.go:121 +0x99
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).recvMessage(0xc2080b3900, 0xc208065560)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:76 +0x55
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).decodeTypeSequence(0xc2080b3900, 0xc2080fcf00, 0x16)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:140 +0x47
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).DecodeValue(0xc2080b3900, 0x3d1d60, 0xc2080fcf40, 0x16, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:208 +0x192
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).Decode(0xc2080b3900, 0x3d1d60, 0xc2080fcf40, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:185 +0x297
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc20810c300, 0xc2080fcf40, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:402 +0x5e
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).readRequestHeader(0xc2080f6e80, 0xa13780, 0xc20810c300, 0x0, 0x0, 0xc2080fcf40, 0xc20810a000, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:575 +0x97
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).readRequest(0xc2080f6e80, 0xa13780, 0xc20810c300, 0xc2080b3600, 0xc2080fc9a0, 0x3ca9e0, 0x0, 0x0, 0x0, 0x0, ...)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:542 +0xa2
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).ServeCodec(0xc2080f6e80, 0xa13780, 0xc20810c300)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:461 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).ServeConn(0xc2080f6e80, 0xa136a8, 0xc20805c780)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:453 +0x208
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.serve(0xa136a8, 0xc20805c780, 0x5ac490, 0x10, 0x5201c0, 0xc2080fc980)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:146 +0x1f9
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.func·001()
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:102 +0x265
2015/03/16 12:44:51 terraform-provider-heroku: created by github.com/hashicorp/terraform/rpc.(*dispenseServer).ResourceProvider
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:103 +0x107
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 38 [select]:
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/yamux.(*Stream).Read(0xc20805c280, 0xc2080b9000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/stream.go:125 +0x3ab
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).fill(0xc2080f8900)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:97 +0x1ce
2015/03/16 12:44:51 terraform-provider-heroku: bufio.(*Reader).Read(0xc2080f8900, 0xc20810bef0, 0x1, 0x9, 0xc208101858, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:174 +0x26c
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadAtLeast(0xa13620, 0xc2080f8900, 0xc20810bef0, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:298 +0xf1
2015/03/16 12:44:51 terraform-provider-heroku: io.ReadFull(0xa13620, 0xc2080f8900, 0xc20810bef0, 0x1, 0x9, 0x1, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/io/io.go:316 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.decodeUintReader(0xa13620, 0xc2080f8900, 0xc20810bef0, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decode.go:121 +0x99
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).recvMessage(0xc2080b3d80, 0xc208064d80)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:76 +0x55
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).decodeTypeSequence(0xc2080b3d80, 0xc2080bef00, 0x16)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:140 +0x47
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).DecodeValue(0xc2080b3d80, 0x3d1d60, 0xc2080bef40, 0x16, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:208 +0x192
2015/03/16 12:44:51 terraform-provider-heroku: encoding/gob.(*Decoder).Decode(0xc2080b3d80, 0x3d1d60, 0xc2080bef40, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/encoding/gob/decoder.go:185 +0x297
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*gobServerCodec).ReadRequestHeader(0xc20810de00, 0xc2080bef40, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:402 +0x5e
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).readRequestHeader(0xc2081177c0, 0xa13780, 0xc20810de00, 0x0, 0x0, 0xc2080bef40, 0xc20810be00, 0x0, 0x0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:575 +0x97
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).readRequest(0xc2081177c0, 0xa13780, 0xc20810de00, 0xc2080b2900, 0xc2080bf720, 0x3ca8c0, 0x0, 0x0, 0x0, 0x0, ...)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:542 +0xa2
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).ServeCodec(0xc2081177c0, 0xa13780, 0xc20810de00)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:461 +0x6d
2015/03/16 12:44:51 terraform-provider-heroku: net/rpc.(*Server).ServeConn(0xc2081177c0, 0xa136a8, 0xc20805c280)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/rpc/server.go:453 +0x208
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.serve(0xa136a8, 0xc20805c280, 0x5ac490, 0x10, 0x5201c0, 0xc2080bef20)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:146 +0x1f9
2015/03/16 12:44:51 terraform-provider-heroku: github.com/hashicorp/terraform/rpc.func·001()
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:102 +0x265
2015/03/16 12:44:51 terraform-provider-heroku: created by github.com/hashicorp/terraform/rpc.(*dispenseServer).ResourceProvider
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/rpc/server.go:103 +0x107
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 47 [runnable]:
2015/03/16 12:44:51 terraform-provider-heroku: net/http.(*persistConn).readLoop(0xc20808c2c0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:928 +0x9ce
2015/03/16 12:44:51 terraform-provider-heroku: created by net/http.(*Transport).dialConn
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:660 +0xc9f
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 49 [syscall, locked to thread]:
2015/03/16 12:44:51 terraform-provider-heroku: runtime.goexit()
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/runtime/asm_amd64.s:2232 +0x1
2015/03/16 12:44:51 terraform-provider-heroku: 
2015/03/16 12:44:51 terraform-provider-heroku: goroutine 48 [select]:
2015/03/16 12:44:51 terraform-provider-heroku: net/http.(*persistConn).writeLoop(0xc20808c2c0)
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:945 +0x41d
2015/03/16 12:44:51 terraform-provider-heroku: created by net/http.(*Transport).dialConn
2015/03/16 12:44:51 terraform-provider-heroku: 	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:661 +0xcbc
2015/03/16 12:44:51 [ERROR] Error walking 'heroku_app.default': 1 error(s) occurred:

* unexpected EOF
2015/03/16 12:44:51 [INFO] Apply walk complete
2015/03/16 12:44:51 [INFO] Writing backup state to: config/infrastructure/heroku/environments/staging.tfstate.backup
2015/03/16 12:44:51 waiting for all plugin processes to complete...
2015/03/16 12:44:51 [DEBUG] /usr/local/terraform/terraform-provider-heroku: plugin process exited
2015/03/16 12:44:51 [DEBUG] /usr/local/terraform/terraform-provider-aws: plugin process exited

--- a/builtin/providers/heroku/resource_heroku_app.go
+++ b/builtin/providers/heroku/resource_heroku_app.go
@@ -358,14 +358,18 @@ func updateConfigVars(
 	vars := make(map[string]*string)
 
 	for _, v := range o {
-		for k, _ := range v.(map[string]interface{}) {
-			vars[k] = nil
+		if v != nil {
+			for k, _ := range v.(map[string]interface{}) {
+				vars[k] = nil
+			}
 		}
 	}
 	for _, v := range n {
-		for k, v := range v.(map[string]interface{}) {
-			val := v.(string)
-			vars[k] = &val
+		if v != nil {
+			for k, v := range v.(map[string]interface{}) {
+				val := v.(string)
+				vars[k] = &val
+			}
 		}
 	}
 

